planning/behavior_path_start_planner_module/src/start_planner_module.cpp

```diff
@@ -93,26 +93,56 @@ void StartPlannerModule::onFreespacePlannerTimer()
 {
   const ScopedFlag flag(is_freespace_planner_cb_running_);
 
-  if (getCurrentStatus() == ModuleStatus::IDLE) {
+  std::shared_ptr<const PlannerData> local_planner_data{nullptr};
+  std::optional<ModuleStatus> current_status_opt{std::nullopt};
+  std::optional<StartPlannerParameters> parameters_opt{std::nullopt};
+  std::optional<PullOutStatus> pull_out_status_opt{std::nullopt};
+  bool is_stopped;
+
+  // making a local copy of thread sensitive data
+  {
+    std::lock_guard<std::mutex> guard(start_planner_data_mutex_);
+    if (start_planner_data_) {
+      const auto & start_planner_data = start_planner_data_.value();
+      local_planner_data = std::make_shared<PlannerData>(start_planner_data.planner_data);
+      current_status_opt = start_planner_data.current_status;
+      parameters_opt = start_planner_data.parameters;
+      pull_out_status_opt = start_planner_data.main_thread_pull_out_status;
+      is_stopped = start_planner_data.is_stopped;
+    }
+  }
+  // finish copying thread sensitive data
+  if (!local_planner_data || !current_status_opt || !parameters_opt || !pull_out_status_opt) {
```

Comment:
Just a small detail, but could you please make the if(!local_planner_data || !current_status_opt || !parameters_opt || !pull_out_status_opt) condensed with a single boolean? 

like const bool is_data_ready = !local_planner_data || !current_status_opt || !parameters_opt || !pull_out_status_opt; 

In the future if  we need to check for more optional variables the if might become too large and CodeScene is complaining about this line.